driver/src/main/java/org/neo4j/driver/ResultTransformerProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver;
+
+import java.io.Serializable;
+
+@FunctionalInterface
+public interface ResultTransformerProvider<T> extends Serializable {
```

Comment:
I would not introduce that interface and instead of go with `Supplier`.

I see the reason for a new interface (you want to make clear it's a new instance), but this is something one can suggest but not enforce.
Imagine a scenario in which people want to run multiple queries to the transformer.